### PR TITLE
[SHAP] Analyze filler

### PR DESCRIPTION
**Pull request by weiguoz**
Add filler for `ANALYZE`.
This PR is a part of [Implement Analyze SQL in SQLFlow](https://github.com/sql-machine-learning/sqlflow/issues/707).

### CODE REVIEW

File: sql/codegen_analyze.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+type analyzeFiller struct {
+	*connectionConfig
+	Columns []string
+	Label   string
+	// feature names:
```
Comment (Yancey0623) [inline sql/codegen_analyze.go]: Please remove the commented code.

File: sql/codegen_xgboost_test.go [hunk 2]
```diff
@@ -321,13 +326,11 @@ LABEL e INTO model_table;
...
 	a.NoError(e)
-	fts, e := verify(pr, testDB)
-	a.NoError(e)
-	filler, e := newXGBoostFiller(pr, nil, fts, testDB)
+	filler, e := newXGBoostFiller(pr, nil, testDB)
 	a.NoError(e)
 
 	a.True(filler.IsTrain)
-	stdSlct := removeLastSemicolon(strings.Replace(filler.StandardSelect, "\n", " ", -1))
+	stdSlct := trimTailOf(strings.Replace(filler.StandardSelect, "\n", " ", -1), ';')
```
Comment (Yancey0623) [inline sql/codegen_xgboost_test.go]: Can use the standar function [strings.TrimRight](https://golang.org/pkg/strings/#TrimRight)

File: sql/codegen.go [hunk 3]
```diff
@@ -68,7 +68,7 @@ type filler struct {
 	Y                  *featureMeta
 	TableName          string
 	modelConfig
-	connectionConfig
+	*connectionConfig
```
Comment (Yancey0623) [inline sql/codegen.go:71]: Why changing to a pointer?

Comment (weiguoz) [reply sql/codegen.go:71]: To improve reusability, `fillDatabaseInfo()` is renamed to `newConnectionConfig()` which returns a pointer.

File: sql/codegen.go [hunk 4]
```diff
@@ -211,56 +211,53 @@ func newFiller(pr *extendedSelect, ds *trainAndValDataset, fts fieldTypes, db *D
 		}
 	}
 
-	return fillDatabaseInfo(r, db)
+	r.connectionConfig, err = newConnectionConfig(db)
+	if err == nil && r.Driver == "hive" {
+		// remove the last ';' which leads to a (hive)ParseException
+		r.TrainingDataset = strings.TrimSuffix(r.TrainingDataset, ";")
```
Comment (typhoonzero) [inline sql/codegen.go]: Maybe rename `TrainingDataset` to `TrainingDatasetSQL` is more meaningful

**Review by sperlingxx — DISMISSED** [review]
LGTM if ci is green

File: sql/codegen_xgboost_test.go (hunk 2 above)
Comment (weiguoz) [reply sql/codegen_xgboost_test.go]: Thanks for your reminder. `trimTailOf` is replaced by [strings.TrimSuffix](https://golang.org/pkg/strings/#TrimSuffix).

File: sql/codegen.go (hunk 4 above)
Comment (weiguoz) [reply sql/codegen.go]: Good idea.

**Review by sperlingxx — APPROVED** [review]
LGTM

**Pull request merged**